### DISCUSSION TITLE
How to use templates for post incidents review.

### DISCUSSION THREAD

**Discussion by manireddyk** [Netflix/dispatch#3494] [Q&A]
We would like to add post incident review to track about incident,  review template is suitable for our need.
we have created review template like this, what should be given in the weblink ?

<img width="1523" alt="Screen Shot 2023-06-12 at 13 49 26" src="https://github.com/Netflix/dispatch/assets/46611593/fcda1034-3def-44c5-913b-e37c04c6afa9">

when we created incident and mark it stable, it doesnt create any template as mentioned in the artifacts. When can I see a copy of template for my incident.
```
 Review Template
A copy of this template is automatically created when an incident is marked as stable. It is used by the incident commander and participants for reconciling any incident learnings, discussions, or post-incident tasks.
```

@kevgliss @wssheldon can u please elaborate how to use template in dispatch

**Comment by mvilanova** [comment]
Can you populate the external id field with the document id and try again?

**Comment by manireddyk** [comment]
@mvilanova  thanks for ur repsonse

sure but where can we see the document generated? the google drive document link that we are giving is like below. but where we can we see the document generated and how it works? where can we write the incident review and how can we check that?

<img width="1270" alt="Screen Shot 2023-06-12 at 14 13 39" src="https://github.com/Netflix/dispatch/assets/46611593/8896b58f-3df3-4dcf-b1cf-e5b87713e976">
I have passed my drive sheet url(open) like this, please correct me If Im wrong
<img width="1288" alt="Screen Shot 2023-06-12 at 14 14 32" src="https://github.com/Netflix/dispatch/assets/46611593/1f445451-ae33-47a6-8187-b60917533dce">

can u also please elaborate on the template usage for incident types

**Comment by mvilanova** [comment]
The document weblink was correct in your first screenshot, not in the second one. Weblink is the full link, External id just the id. The document will be copied to the incident folder where the other docs are.

**Comment by csetarun2** [comment]
hi @mvilanova, can you please guide where the incident folder resides, u mean it in the UI or in the filesystem?

**Comment by csetarun2** [comment]
Hi @mvilanova can you please help on this..

Where can I find those created docs...should I enable gdribe plugin for this ?

**Comment by mvilanova** [comment]
Yes, you need to enable and configure the `Google Drive Plugin - Storage Management` plugin. Once you do that, Dispatch will create a Google drive folder for each incident and make copies of the templates there.

**Comment by csetarun2** [comment]
Hi @mvilanova ,
I have created service account in gcloud created json key, and provided those values, please correct me If im wrong 
& can you please help how to create developer key and is workspace domain is my domain address right ?


<img width="171" alt="image" src="https://github.com/Netflix/dispatch/assets/21236859/b0a9aed0-c578-4c6d-b756-a879fb790047">

**Reply by mvilanova** [reply]
The root folder id needs to be the folder id not the URL to the folder

**Reply by mvilanova** [reply]
You can create a developer key in your GCP console under credentials. There's documentation online on how to do it.

**Reply by mvilanova** [reply]
We explain how here: https://netflix.github.io/dispatch/docs/administration/settings/plugins/configuring-g-suite#enable-required-apis-in-google-cloud-platform

**Comment by csetarun2** [comment]
@mvilanova @kevgliss any suggestion/document on thus

**Comment by manireddyk** [comment]
Even I got struck at the same step as well. Thank you for bringing it up @csetarun2

**Comment by csetarun2** [comment]
Hi @manireddyk check whether u enabled both docs and storage plugin also check the proper link ids, permissions...enable both drive and doc plugins in your google project..

Try to check logs..if u make mistake it will throw some errors...u can troubleshoot those
